src/main/java/randoop/generation/ForwardGenerator.java

```diff
@@ -270,6 +279,38 @@ public Set<Sequence> getAllSequences() {
     return this.allSequences;
   }
 
+  /**
+   * If the dynamic type (the run-time class) of the sequence's output (the value returned by the
+   * last statement) is a subtype of its static type, cast it to its dynamic type. This allows
+   * Randoop to call methods on it that do not exist in the supertype.
+   *
+   * <p>This implements the "GRT Elephant-Brain" component, as described in "GRT:
+   * Program-Analysis-Guided Random Testing" by Ma et. al (ASE 2015):
+   * https://people.kth.se/~artho/papers/lei-ase2015.pdf.
+   *
+   * @param eSeq an executable sequence; may be side-effected
+   */
+  private void castToRunTimeType(ExecutableSequence eSeq) {
+    List<ReferenceValue> lastValues = eSeq.getLastStatementValues();
+    if (lastValues.isEmpty()) {
+      return;
+    }
+    ReferenceValue lastValue = lastValues.get(0);
+    Type declaredType = lastValue.getType();
+    Type runTimeType = Type.forClass(lastValue.getObjectValue().getClass());
+
+    if (!runTimeType.equals(declaredType) && runTimeType.isSubtypeOf(declaredType)) {
+      TypedOperation castOperation = TypedOperation.createCast(declaredType, runTimeType);
+
+      // Get the first variable of the last statement and cast it to the run-time type.
+      Variable variable = eSeq.sequence.firstVariableForTypeLastStatement(declaredType, false);
+
+      if (variable != null) {
```

Comment:
How can `variable` be null?  If it is null, does that indicate a bug in Elephant Brain?  If so, please make it an assertion rather than an `if` condition.